applications/callflow/src/cf_route_win.erl

```diff
@@ -375,6 +374,23 @@ get_incoming_security(Call) ->
              )
     end.
 
+-spec get_endpoint_id(kapps_call:call()) ->kz_term:api_ne_binary().
+get_endpoint_id(Call) ->
+    DefaultEndpointId = kapps_call:authorizing_id(Call),
+    kapps_call:kvs_fetch(?RESTRICTED_ENDPOINT_KEY, DefaultEndpointId, Call).
+
+-spec include_denied_call_restrictions(kapps_call:call()) -> kapps_call:call().
+include_denied_call_restrictions(Call) ->
+    case kz_endpoint:get(get_endpoint_id(Call), Call) of
+        {'error', _R} ->
+            Call;
+        {'ok', JObj} ->
+            CallRestriction = kz_json:get_value(<<"call_restriction">>, JObj),
```

Comment:
Use the more specific getters when you can: `get_json_value/2` (maybe with a default `kz_json:new()`?) and `get_ne_binary_value/2` on the next line.